diff --git a/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js b/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js
index 32a422a..162a1bd 100644
--- a/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js
+++ b/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js
@@ -177,7 +177,8 @@ export function NativeBottomTabView({
             route
           }),
           tabKey: route.key,
-          icon: icon,
+          icon: icon?.ios ?? icon?.shared,
+          iconResourceName: icon?.android?.name,
           selectedIcon: selectedIcon?.ios ?? selectedIcon?.shared,
           tabBarItemBadgeBackgroundColor: badgeBackgroundColor,
           tabBarItemBadgeTextColor: badgeTextColor,
@@ -215,7 +216,7 @@ export function NativeBottomTabView({
 }
 function getPlatformIcon(icon) {
   return {
-    ios: icon?.type === 'sfSymbol' ? icon : icon?.type === 'image' && icon.tinted !== false ? {
+    ios: icon?.type === 'sfSymbol' ? { sfSymbolName: icon.name } : icon?.type === 'image' && icon.tinted !== false ? {
       type: 'templateSource',
       templateSource: icon.source
     } : undefined,
diff --git a/node_modules/@react-navigation/bottom-tabs/src/unstable/NativeBottomTabView.native.tsx b/node_modules/@react-navigation/bottom-tabs/src/unstable/NativeBottomTabView.native.tsx
index 5cc1429..bf6c7e0 100644
--- a/node_modules/@react-navigation/bottom-tabs/src/unstable/NativeBottomTabView.native.tsx
+++ b/node_modules/@react-navigation/bottom-tabs/src/unstable/NativeBottomTabView.native.tsx
@@ -241,7 +241,8 @@ export function NativeBottomTabView({ state, navigation, descriptors }: Props) {
               onDidDisappear={() => onTransitionEnd({ closing: true, route })}
               key={route.key}
               tabKey={route.key}
-              icon={icon}
+              icon={icon?.ios ?? icon?.shared}
+              iconResourceName={icon?.android?.name}
               selectedIcon={selectedIcon?.ios ?? selectedIcon?.shared}
               tabBarItemBadgeBackgroundColor={badgeBackgroundColor}
               tabBarItemBadgeTextColor={badgeTextColor}
@@ -288,7 +289,7 @@ function getPlatformIcon(icon: Icon): PlatformIcon {
   return {
     ios:
       icon?.type === 'sfSymbol'
-        ? icon
+        ? { sfSymbolName: icon.name } as any
         : icon?.type === 'image' && icon.tinted !== false
           ? {
               type: 'templateSource',
